server.go

```diff
@@ -258,6 +263,16 @@ func (s *ExtensionManagerServer) Shutdown(ctx context.Context) error {
 		}()
 	}
 
+	// Also close the client that is attached to the osquery hosted pipe
+	if s.serverClient != nil {
+		serverClient := s.serverClient
+		s.serverClient = nil
+
+		go func() {
```

Comment:
Can you explain why this happens in a separate goroutine rather than synchronously in this function?

Reply:
I just tested it without the routine and it worked fine - removed the routine now.
I originally just copied the flow as was happening with the server above.